File: frontend/src/api/presets.ts
import { apiFetch } from "./client";

export type PresetField = {
  name: string;
  label: string;
  type: "string" | "number" | "boolean";
  required: boolean;
  default?: string | number | boolean | null;
  enum?: Array<string | number>;
};

export type Preset = {
  id: string;
  label: string;
  job_type: string;
  network_id: string;
  price_rub: number;
  eta_seconds?: number | null;
  poll_interval_seconds?: number | null;
  timeout_seconds?: number | null;
  fields: PresetField[];
};

export async function getPresets() {
  return apiFetch<{ items: Preset[] }>("/presets");
}

File: backend/app/core/schemas/presets.py
from typing import Any
from pydantic import BaseModel


class PresetField(BaseModel):
    name: str
    label: str
    type: str
    required: bool
    default: Any | None = None
    enum: list[Any] | None = None


class PresetOut(BaseModel):
    id: str
    label: str
    job_type: str
    network_id: str
    eta_seconds: int | None = None
    poll_interval_seconds: float | None = None
    timeout_seconds: int | None = None
    fields: list[PresetField]


class PresetList(BaseModel):
    items: list[PresetOut]
